packages/nodes-base/nodes/Slack/SlackTrigger.node.ts

```diff
@@ -0,0 +1,415 @@
+					},
+				],
+			},
+		],
+	};
+
+	methods = {
+		listSearch: {
+			async getChannels(
+				this: ILoadOptionsFunctions,
+				filter?: string,
+			): Promise<INodeListSearchResult> {
+				const qs = { types: 'public_channel,private_channel', limit: 1000 };
+				const channels = (await slackApiRequestAllItems.call(
+					this,
+					'channels',
+					'GET',
+					'/conversations.list',
+					{},
+					qs,
+				)) as Array<{ id: string; name: string }>;
+				const results: INodeListSearchItems[] = channels
+					.map((c) => ({
+						name: c.name,
+						value: c.id,
+					}))
+					.filter(
+						(c) =>
+							!filter ||
+							c.name.toLowerCase().includes(filter.toLowerCase()) ||
+							c.value?.toString() === filter,
+					)
+					.sort((a, b) => {
+						if (a.name.toLowerCase() < b.name.toLowerCase()) return -1;
+						if (a.name.toLowerCase() > b.name.toLowerCase()) return 1;
+						return 0;
+					});
+				return { results };
+			},
+		},
+		loadOptions: {
+			async getUsers(this: ILoadOptionsFunctions): Promise<INodePropertyOptions[]> {
+				const returnData: INodePropertyOptions[] = [];
+				const users = (await slackApiRequestAllItems.call(
+					this,
+					'members',
+					'GET',
+					'/users.list',
+				)) as Array<{ id: string; name: string }>;
+				for (const user of users) {
+					returnData.push({
+						name: user.name,
+						value: user.id,
+					});
+				}
+
+				returnData.sort((a, b) => {
+					if (a.name < b.name) {
+						return -1;
+					}
+					if (a.name > b.name) {
+						return 1;
+					}
+					return 0;
+				});
+
+				return returnData;
+			},
+		},
+	};
+
+	webhookMethods = {
+		default: {
+			async checkExists(this: IHookFunctions): Promise<boolean> {
+				return true;
+			},
+			async create(this: IHookFunctions): Promise<boolean> {
+				return true;
+			},
+			async delete(this: IHookFunctions): Promise<boolean> {
+				return true;
+			},
+		},
+	};
+
+	async webhook(this: IWebhookFunctions): Promise<IWebhookResponseData> {
+		const filters = this.getNodeParameter('trigger', []) as string[];
+		const req = this.getRequestObject();
+		const options = this.getNodeParameter('options', {}) as IDataObject;
+		const binaryData: IBinaryKeyData = {};
+		const watchWorkspace = this.getNodeParameter('watchWorkspace', false) as boolean;
+
+		// Check if the request is a challenge request
+		if (req.body.type === 'url_verification') {
+			const res = this.getResponseObject();
+			res.status(200).json({ challenge: req.body.challenge }).end();
+
+			return {
+				noWebhookResponse: true,
+			};
+		}
+
+		// Check if the event type is in the filters
+		const eventType = req.body.event.type as string;
+
+		if (
+			eventType === 'message' &&
+			!filters.includes('file_share') &&
+			!filters.includes('any_event')
+		) {
+			return {};
+		}
+
+		if (!watchWorkspace) {
+			if (
+				req.body.event.channel !==
```

Comment:
I found that channel is not always at this location depending on the event:

For example for the `reaction_added` event, it's at `event.item.channel`, there might be more

Reply:
Solid find, as I use both later in the workflow I have set a const to one of these 2 options as I think they were the only ones I found.